test/unit/test_factory.cpp

```diff
@@ -2103,6 +2103,58 @@ TEST(
     }
 }
 
+TEST(factory, AuthorityFactory_getAvailableGeoidmodels) {
+    const std::list<std::string> navd88geoidmodels{
+        "GEOID03",  "GEOID06", "GEOID09", "GEOID12A",
+        "GEOID12B", "GEOID18", "GEOID99"};
+
+    const std::string OSGM15{"OSGM15"};
+
+    auto checkNavd88 = [&navd88geoidmodels](const std::list<std::string> &res) {
+        ASSERT_EQ(res.size(), navd88geoidmodels.size());
+        auto it = navd88geoidmodels.cbegin();
+        for (const auto &model : res) {
+            EXPECT_EQ(*it, model);
+            it++;
+        }
+    };
+    auto factory =
+        AuthorityFactory::create(DatabaseContext::create(), std::string());
```

Comment:
In case we get objects under another authority with same codes...
```suggestion
        AuthorityFactory::create(DatabaseContext::create(), "EPSG");
```